flow/core/rewards.py

```diff
@@ -469,24 +468,10 @@ def instantaneous_power(env, veh_ids=None, gain=.001):
         veh_ids = [veh_ids]
     for veh_id in veh_ids:
         speed = env.k.vehicle.get_speed(veh_id)
-        prev_speed = env.k.vehicle.get_previous_speed(veh_id)
-        # negative acceleration indicates regenerative breaking for some energy models
-        accel = (speed - prev_speed) / env.sim_step
+        accel = env.k.vehicle.get_accel_no_noise_with_failsafe(veh_id)
         grade = 0
-
-        parameter = None
-        # determine which energy model to use
-        if veh_id is "Prius":
-            energy_model = env.k.PriusEnergy
-            initial_soc = 0.9
-            parameter = initial_soc
-        elif veh_id is "Tacoma":
-            energy_model = env.k.TacomaEnergy
-        elif veh_id is "EV":
-            energy_model = env.k.PDMElectric
-        else:
-            energy_model = env.k.PDMCombustionEngine
-        
-        inst_power = energy_model.get_instantaneous_power(parameter, accel, speed, grade)
+        # need to pass the parameter value to the init of the energy models ?????
+        inst_power = energy_model(accel, speed, grade)
```

Comment:
so when you get here, the class should already be instantiated for the vehicle, and you just need to access the method within the class, so:
`inst_power = energy_classes[veh_id].get_instantaneous_power(accel, speed, grade)`